src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+    private static final String byeMessage = "Bye. Hope to see you again soon!";
+
+    private static final String listMessage = "Here are the tasks in your list:";
+
+    private static final int MAXIMUM_TASKS = 100;
+
+    private static boolean shouldContinue;
+
+    private static String command;
+
+    private static String action;
+
+    private static Scanner input;
+
+    private static Task[] instructions;
+
+    private static int count;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        initDuke();
+        displayWelcome();
+        while (shouldContinue) {
+            readInput();
+            setAction();
+            processCommand();
+        }
+    }
+
+    private static void processCommand() {
+        switch(action) {
+        case "bye":
+            sayBye();
+            break;
+        case "list":
+            listTasks();
+            break;
+        case "done":
+            completeTask();
+            break;
+        case "deadline":
+            setDeadline();
+            confirmTask();
+            break;
+        case "event":
+            createEvent();
+            confirmTask();
+            break;
+        case "todo":
+            addTodo();
+            confirmTask();
+            break;
+        }
+    }
+
+    private static void addTodo() {
+        String description = command.substring(5);
+        instructions[count] = new Todo(description);
+        count += 1;
+    }
+
+    private static void createEvent() {
+        int index = command.indexOf("/at");
+        String description = command.substring(6, index - 1);
+        String duration = command.substring(index + 4);
+        instructions[count] = new Event(description, duration);
+        count += 1;
+    }
+
+    private static void setDeadline() {
+        int index = command.indexOf("/by");
+        String description = command.substring(9, index - 1);
+        String by = command.substring(index + 4);
+        instructions[count] = new Deadline(description, by);
+        count += 1;
+    }
+    private static void completeTask() {
+        String[] phrases = command.split(" ");
+        int index = Integer.parseInt(phrases[1]);
+        instructions[index-1].markAsDone();
+        System.out.println(completeMessage);
+        System.out.println("  " + instructions[index-1]);
+    }
+
+    private static void setAction() {
+        String[] phrases = command.split(" ");
+        action = phrases[0];
+    }
+
+    private static void listTasks() {
+        System.out.println(listMessage);
+        for (int i = 0; i < count; i++) {
+            System.out.println((i+1) + "." + instructions[i]);
+        }
+    }
+
+    private static void sayBye() {
```

Comment:
Consider cutting down on static methods if possible. Replace them with static variables. Static methods are essentially "global" and might interfere with SOLID principles

Reply:
I guess I tried to apply the "extract method" to almost everything. Will use static variables next time!